### PR TITLE
reset_config instead of update_config in tests

### PR DESCRIPTION
**Pull request by tomsilver**
this will hopefully put an end to the nightmare of unit tests interfering with each other. we should now use `reset_config` instead of `update_config` in all tests.

I went through all of the tests that previously used `update_config` and removed superfluous things like `seed: 123` (which happens automatically), multiple consecutive calls to `update_config` to deal with the `get_arg_specific_settings` stuff, and calls to `update_config` at the end of unit tests.

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -1267,16 +1268,34 @@ def update_config(args: Dict[str, Any], default_seed: int = 123) -> None:
             # pass in a value and this key is already in the
             # configuration dict, add the current value to args.
             args[k] = getattr(CFG, k)
-    # Maintain the invariant that CFG has some seed and some
-    # experiment_id set. This is very useful in unit tests, where
-    # there are often no command line args being passed.
-    args["seed"] = args.get("seed", default_seed)
-    args["experiment_id"] = args.get("experiment_id", "")
```
Comment (ronuchit) [inline src/utils.py:1274]: Where is the default setting of experiment_id happening now?

File: src/utils.py [hunk 2]
```diff
@@ -1267,16 +1268,34 @@ def update_config(args: Dict[str, Any], default_seed: int = 123) -> None:
...
+        "--env",
+        "default env placeholder",
+        "--seed",
+        str(default_seed),
+        "--approach",
+        "default approach placeholder",
+    ])
+    arg_dict = {
+        k: v
+        for k, v in GlobalSettings.__dict__.items() if not k.startswith("_")
```
Comment (ronuchit) [inline src/utils.py:1292]: Will this work correctly with arg-specific settings?

File: tests/approaches/test_base_approach.py [hunk 3]
```diff
@@ -28,9 +28,7 @@ def _solve(self, task: Task, timeout: int) -> Callable[[State], Action]:
 
 def test_base_approach():
     """Tests for BaseApproach class."""
-    utils.update_config({
-        "seed": 123,
-    })
+    utils.reset_config({})
```
Comment (ronuchit) [inline tests/approaches/test_base_approach.py:31]: Why do we need calls like this? Does it just set you back to the default CFG that's in settings.py? If so, maybe we should make the argument be optional (default None) and turn calls like this into `utils.reset_config()`.

**Review by ronuchit — APPROVED** [review]
thank you so much 🙏 🙏 🙏  this is amazing

File: tests/approaches/test_base_approach.py (hunk 3 above)
Comment (ronuchit) [reply tests/approaches/test_base_approach.py:31]: should be quick to do with a grep

File: src/utils.py (hunk 1 above)
Comment (tomsilver) [reply src/utils.py:1274]: it defaults to an empty str in args.py

File: src/utils.py (hunk 2 above)
Comment (tomsilver) [reply src/utils.py:1292]: yeah, the arg-specific settings are added in the call to `update_config` at the end of this function

**Pull request merged**